### ISSUE TITLE
Error in imzml export

### ISSUE THREAD

**Issue by liuxiawei** [kuwisdelu/Cardinal#49]
I use Cardinal to treat MSI data and save as imzml file. When I read imzml by pyimzml, An strange error report.
In imzml, the data contain  the number displayed as 1.156e+09. I think it is a bug caused by writing by R.
![image](https://github.com/user-attachments/assets/df836cbb-2544-4f5e-add4-24a3eec325d3)

**Comment by kuwisdelu** [comment]
Can you please provide the information described in #41 ?

**Comment by liuxiawei** [comment]
The `Cardinal_3.6.5` manually install from github last month. 
```
R version 4.3.1 (2023-06-16)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/liuxiawei/micromamba/envs/cardinal3/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Asia/Shanghai
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Cardinal_3.6.5      S4Vectors_0.40.2    BiocParallel_1.36.0 BiocGenerics_0.48.1 ProtGenerics_1.34.0

loaded via a namespace (and not attached):
 [1] nlme_3.1-163       cli_3.6.1          rlang_1.1.1        DBI_1.1.3          tiff_0.1-11        png_0.1-8          RCurl_1.98-1.12    htmltools_0.5.6.1  ontologyIndex_2.11
[10] Biobase_2.62.0     locfit_1.5-9.8     grid_4.3.1         abind_1.4-5        bitops_1.0-7       fastmap_1.1.1      compiler_4.3.1     codetools_0.2-19   CardinalIO_1.0.0  
[19] irlba_2.3.5.1      htmlwidgets_1.6.2  fftwtools_0.9-11   EBImage_4.44.0     lattice_0.22-5     digest_0.6.33      matter_2.6.3       parallel_4.3.1     Matrix_1.6-1.1    
[28] tools_4.3.1        jpeg_0.1-10        biglm_0.9-2.1     
```

**Comment by kuwisdelu** [comment]
This should now be fixed by CardinalIO 1.2.2 and 1.3.6 (https://github.com/kuwisdelu/CardinalIO/commit/eb1ef104e5e6b6597639cd2d1ea8dac87c0ff437).

The Bioconductor 3.19 release branch is already frozen to prepare for the 3.20 release, but the updated version is installable by:

```
remotes::install_github("kuwisdelu/CardinalIO", ref=remotes::github_release())
```

CardinalIO 1.4 will include the bug fix and will release with Bioconductor 3.20 next week.

Note that this should have only happened when it could be done without loss of precision, so it should only have affected parsing, and you should be able to do a find and replace of "1.156e+09" with "1156000000" if that's the only offset affected to fix it without rewriting the whole file.

**Comment by liuxiawei** [comment]
Thanks for this great work and continue working！
